tensorboard/webapp/runs/views/runs_table/runs_table_container.ts

```diff
@@ -519,6 +569,34 @@ export class RunsTableContainer implements OnInit, OnDestroy {
     return slicedItems;
   }
 
+  private getRunTableDataForExperiment(
+    experimentId: string
+  ): Observable<TableData[]> {
+    return combineLatest([
+      this.store.select(getRuns, {experimentId}),
+      this.store.select(getRunColorMap),
+    ]).pipe(
+      map(([runs, colorMap]) => {
+        return runs.map((run) => {
+          const tableData: TableData = {
+            id: run.id,
+            color: colorMap[run.id],
+          };
+          this.runsColumns.forEach((column) => {
+            switch (column.type) {
+              case ColumnHeaderType.RUN:
+                tableData[column.name!] = run.name;
+                break;
+              default:
+                break;
+            }
+          });
+          return tableData;
+        });
+      })
+    );
+  }
```

Comment:
This should be a selector, but I don't mind it being this way for now (until I get my generalized selector checked in).

Could this at least be a lot shorter?
```suggestion
  const runColumnName = this.runColumns.find((column) => column.type === ColumnHeaderType.RUN).name;
  return runs.map((run) => ({
    id: run.id,
    color: colorMap[run.id],
    [runColumnName]: run.name
  });
```